react/react_forms/README.md

```diff
@@ -1,387 +1,477 @@
+### Handling submission
 
-Since we've done everything in a `controlled` manner, we have all the values that we need in our component's state. So all we have to do is modify our `onSubmit` function to grab the variables.
+Since we've done everything in a controlled manner, we have all the values that we need in our component's state. All we have to do now is modify our `onSubmit` function to get these values from our state and do what we want with them.
 
-When the submit button is pressed, `handleFormSubmit` is called, and reads from the state to ensure that the form can be submitted. In a future lesson, we'll see how to react to the state changing to ensure that the user isn't able submit the form until all fields are valid.
+When the `submit` button is pressed, `handleFormSubmit` is called, and reads from the state to ensure that the form can be submitted. In a future lesson, we'll see how to react to the state changing to ensure that the user isn't able submit the form until all fields are valid.
 
 ```jsx
-const allFieldsValid = () => {
-  return (
-    notARobot &&
-    title &&
-    firstName &&
-    lastName
-  );
-};
+allFieldsValid = () => {
+  const { notARobot, userTitle, firstName, lastName } = this.state;
+  return notARobot && userTitle && firstName && lastName;
+}
 
-const handleFormSubmit = (event) => {
+handleFormSubmit = (event) => {
   event.preventDefault();
-  if(allFieldsValid()) {
+  if (this.allFieldsValid()) {
     alert("Form submitted!");
-  }
-  else {
+  } else {
     alert("Please fill out the form completely")
   }
 }
 ```
 
-Normally on submit we would probably send a POST request to the server with the form data. But for this example, we just pop up an alert.
+Normally, `handleFormSubmit` is where we would send the form data to our server via an AJAX. For this example, however, we'll just validate that our data is truthy using the helper function `allFieldsValid`. If our data is valid, we pop up an alert to let them know the form has been successfully submitted.
 
-We can also show the form data in the alert, just to demonstrate that we have access to all the data in the form.
+We can also show the form data in the alert, just to demonstrate that we have access to all the data in the form:
 
 ```js
-const handleFormSubmit = (event) => {
+handleFormSubmit = (event) => {
   event.preventDefault();
-  if(allFieldsValid()) {
-    alert(`Form submitted! \n ${notARobot} \n ${title} \n ${firstName} \n ${lastName}`);
-  }
-  else {
-    alert("Please fill out the form completely")
+  const { notARobot, userTitle, firstName, lastName } = this.state;
+
+  if (this.allFieldsValid()) {
+    alert(
+      `Form submitted! \n ${notARobot} \n ${userTitle} \n ${firstName} ${lastName}`
+    );
+  } else {
+    alert("Please fill out the form completely");
   }
-}
+};
 ```
 
-> You can see the final product in the `solution` branch on the example repo.
-
-## Uncontrolled components
+## Uncontrolled Components
```

Comment:
@mikeboyle are we going to go into uncontrolled components? I thought we decided on just controlled with links to additional reading? 

Also curious if we do go down the uncontrolled route , shouldn't we be using refs ? https://reactjs.org/docs/uncontrolled-components.html

Reply:
Good catch - @crymall you can cut the entire section on uncontrolled components